app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -94,18 +98,20 @@ class SettingsFragment : Fragment() {
      * See the comment at the declaration of these views in XML for more details.
      */
     private fun updateForAccessibility() {
-        // When VoiceView is enabled, since the parent is initially focusable in the XML, focus is
-        // given to the parent when the Settings opens. Here, after we set focusable to false, we
-        // must also explicitly clear focus in order to give focus to the child.
+        // In order to read Accessibility text for the Telemetry checkbox WITH checked state,
+        // we need to focus the checkbox in VoiceView instead of the containing view.
         //
         // When we change VoiceView from enabled -> disabled and this setting is focused, focus is
         // cleared from this setting and nothing is selected. This is fine: the user can press
         // left-right to focus something else and it's an edge case that I don't think it is worth
         // adding code to fix.
         val context = context ?: return
-        val shouldFocus = context.isVoiceViewEnabled()
-        telemetryView.isFocusable = shouldFocus
-        if (!shouldFocus) { telemetryView.clearFocus() }
+        val shouldFocusButton = context.isVoiceViewEnabled()
+        telemetryView.isFocusable = !shouldFocusButton
+        // Clear focus so that focus passes to child telemetryButton view.
+        if (shouldFocusButton) {
+            telemetryView.clearFocus()
```

Comment:
Optional nit: this might be more clear if we broke it into a few inner functions, even if it introduces a little bit of code repetition.  Maybe something like this?

```kotlin
fun focusView() = telemetryView.isFocusable = true
fun focusButton() {
  // Clear focus so that focus passes to child telemetryButton view.
  telemetryView.isFocusable = false
  telemetryView.clearFocus
}
when {
  shouldFocusButton -> focusButton()
  else -> focusView()
}
```